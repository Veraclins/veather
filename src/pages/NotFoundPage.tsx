import React from 'react';
import { ReactComponent as NotFound } from 'assets/404.svg';
import { useHistory } from 'react-router';

const NotFoundPage: React.FC = () => {
  const history = useHistory();

  const gotoHomePage = () => {
    history.push('/');
  };

  return (
    <div className="contained">
      <NotFound className="max-w-90" />
      <h1 className="mt-4 text-center">Something&apos;s missing</h1>
      <p className="text-center">
        This page is missing or you assembled the link incorrectly.
      </p>
      <button
        onClick={gotoHomePage}
        className="text-no-transform p-2-5 px-3 mt-3 bg-mast-theme-deep text-white"
      >
        Go to Home Page
      </button>
    </div>
  );
};

export default NotFoundPage;
