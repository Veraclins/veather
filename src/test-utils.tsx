import { render, RenderOptions } from '@testing-library/react';

import WeatherProvider from 'context/WeatherProvider';

const AllTheProviders: React.FC = ({ children }) => {
  return <WeatherProvider>{children}</WeatherProvider>;
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
