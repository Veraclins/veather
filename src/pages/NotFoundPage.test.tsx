import { render } from 'test-utils';
import NotFoundPage from 'pages/NotFoundPage';

describe('NotFoundPage Page', () => {
  it('renders the not found page', () => {
    const view = render(<NotFoundPage />);
    expect(view).toMatchSnapshot();
  });
});
