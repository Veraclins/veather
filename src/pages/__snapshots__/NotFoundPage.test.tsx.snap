// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage Page renders the not found page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="contained"
      >
        <svg>
          404.svg
        </svg>
        <h1
          class="mt-4 text-center"
        >
          Something's missing
        </h1>
        <p
          class="text-center"
        >
          This page is missing or you assembled the link incorrectly.
        </p>
        <button
          class="bg-blue p-2-5 px-3 mt-3 text-white"
          data-testid="goto-home"
        >
          Let's take you home
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="contained"
    >
      <svg>
        404.svg
      </svg>
      <h1
        class="mt-4 text-center"
      >
        Something's missing
      </h1>
      <p
        class="text-center"
      >
        This page is missing or you assembled the link incorrectly.
      </p>
      <button
        class="bg-blue p-2-5 px-3 mt-3 text-white"
        data-testid="goto-home"
      >
        Let's take you home
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
